### PR TITLE
Add support for SOF with Zephyr for i.MX8QXP board

### PR DESCRIPTION
**Pull request by iuliana-prodan**
Add support for i.MX8QXP board by defining defconfig,
board configuration and device tree.
This is very similar with i.MX8QM board and uses the
same SoC and dts/xtensa/nxp/nxp_imx8.dtsi.

From Audio DSP point of view, i.MX8QXP has a
1 Xtensa HiFi4 core, with 64 KB TCM,
448 KB OCRAM, 8MB SDRAM and 1 ESAI, 1 SAI as
audio interfaces.

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

**Review by dbaluta — APPROVED** [review]

**Review by paulstelian97 — APPROVED** [review]

File: boards/xtensa/nxp_adsp_imx8x/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_NXP_ADSP_IMX8X
+
+config BOARD
+	default "nxp_adsp_imx8x"
+
+config DUMMY_DMA
+	default y
+	depends on DMA
```
Comment (tejlmand) [inline boards/xtensa/nxp_adsp_imx8x/Kconfig.defconfig:12]: what's the purpose of this ?

I can't find it's usage or description on where this exists in Kconfig or why.

File: boards/xtensa/nxp_adsp_imx8x/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+config IMX_EDMA
+	default y
+	depends on DMA
+
+
```
Comment (tejlmand) [inline boards/xtensa/nxp_adsp_imx8x/Kconfig.defconfig:18]: minot nit: why the extra line ?
```suggestion
```

**Review by tejlmand — CHANGES_REQUESTED** [review]
What's the purpose of `DUMMY_DMA`?
I can't find it elsewhere in our Kconfig tree.

File: boards/xtensa/nxp_adsp_imx8x/Kconfig.defconfig (hunk 1 above)
Comment (iuliana-prodan) [reply boards/xtensa/nxp_adsp_imx8x/Kconfig.defconfig:12]: This is the software DMA driver used by i.MX and is defined in SOF - https://github.com/zephyrproject-rtos/sof/blob/zephyr/src/drivers/Kconfig#L11. 
We need this for the communication between the host and DSP.

File: boards/xtensa/nxp_adsp_imx8x/Kconfig.defconfig (hunk 2 above)
Comment (iuliana-prodan) [reply boards/xtensa/nxp_adsp_imx8x/Kconfig.defconfig]: Right, no need for the extra line - I've removed it.

File: boards/xtensa/nxp_adsp_imx8x/Kconfig.defconfig (hunk 1 above)
Comment (tejlmand) [reply boards/xtensa/nxp_adsp_imx8x/Kconfig.defconfig:12]: thanks.

**Review by tejlmand — APPROVED** [review]

**Pull request merged**